framework/src/base/NeighborCoupleable.C

```diff
@@ -32,9 +32,9 @@ NeighborCoupleable::~NeighborCoupleable() {}
 const VariableValue &
 NeighborCoupleable::coupledNeighborValue(const std::string & var_name, unsigned int comp)
 {
-  MooseVariable * var = getVar(var_name, comp);
+  MooseVariable * var = dynamic_cast<MooseVariable *>(getVar(var_name, comp));
```

Comment:
Are we sure this will be always not NULL?

Reply:
Can we have an API that will give back `MooseVariable *` that will be checked for `!nullptr` and in case if `nullptr`, we get an error message. It would be a lot better than to have not-checked `dynamic_cast`s.

Reply:
`Coupleable::getVar` does this but these eluded my search and destroy of unnecessary casts

Reply:
Right, I missed this was in `NeighborCoupleable`. Easy fix then. 👍